lib/plugins/aws/package/compile/functions.js

```diff
@@ -653,30 +659,42 @@ class AwsCompileFunctions {
         .IamRoleLambdaExecution.Properties.Policies[0].PolicyDocument.Statement;
 
     const action = (() => {
-      if (!functionAddress.startsWith('arn:') || functionAddress.includes(':function:')) {
-        return 'lambda:InvokeFunction';
+      if (typeof functionAddress === 'string') {
+        if (!functionAddress.startsWith('arn:') || functionAddress.includes(':function:')) {
+          return 'lambda:InvokeFunction';
+        }
+        if (functionAddress.includes(':sqs:')) return 'sqs:SendMessage';
+        if (functionAddress.includes(':sns:')) return 'sns:Publish';
+        if (functionAddress.includes(':event-bus/')) return 'events:PutEvents';
+      }
+      if (typeof functionAddress === 'object' && 'Ref' in functionAddress) {
+        const type = this.serverless.service.resources.Resources[functionAddress.Ref].Type;
```

Comment:
This would first need to be first validated that `Ref` is actually referencing a resource that is defined in CloudFormation and I'm afraid that at this point the resources might not yet be available fully to be checked. Could you please validate that with a test and introduce a check that throws a meaningful error? Maybe the better way would be to explicitly require users to provide a type in such case? Trying to resolve such values from references is always tricky

Reply:
Thanks for the proposal @ALOHACREPES345 - I think it would be easier to configure with such approach:

```
onSuccess:
  type: ...
  arn: ...
```

while at the same time, we would still support this syntax:

```
onSuccess: ...
```

but throw a descriptive error if someone tries to pass it as Ref for example. 

We already do something similar e.g. here: https://github.com/serverless/serverless/blob/6a6417c29fe6c06286562ab7604536004bf62334/lib/plugins/aws/package/compile/events/stream.js#L167-L184 where the type can be passed explicitly or it can be derived from passed configuration - in this case we don't throw errors, but I think in case of `destinations` we should keep things simple and just require `type` explicitly if someone passes an ARN

Reply:
Thank you very much for sharing an example. 
I'll take a closer look at the code.